community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/aggregation/SumFunction.scala

```diff
@@ -23,21 +23,32 @@ import org.neo4j.cypher.internal.runtime.interpreted.ExecutionContext
 import org.neo4j.cypher.internal.runtime.interpreted.commands.expressions.Expression
 import org.neo4j.cypher.internal.runtime.interpreted.pipes.QueryState
 import org.neo4j.values.AnyValue
-import org.neo4j.values.storable.{NumberValue, Values}
 import org.neo4j.values.utils.ValueMath.overflowSafeAdd
 
 class SumFunction(val value: Expression)
   extends AggregationFunction
-  with NumericExpressionOnly {
+    with NumericOrDurationAggregationExpression {
 
   def name = "SUM"
 
-  private var sum: NumberValue = Values.ZERO_INT
-  override def result(state: QueryState): AnyValue = sum
+  override def result(state: QueryState): AnyValue = aggregatingType match {
+    case None =>
+      sumNumber
+    case Some(AggregatingNumbers) =>
+      sumNumber
+    case Some(AggregatingDurations) =>
+      sumDuration
+  }
 
   override def apply(data: ExecutionContext, state: QueryState) {
-    actOnNumber(value(data, state), (number) => {
-      sum = overflowSafeAdd(sum, number)
-    })
+    val vl = value(data, state)
+    actOnNumberOrDuration(vl,
+      number => {
+        sumNumber = overflowSafeAdd(sumNumber, number)
+      },
+      duration => {
+        sumDuration = sumDuration.add(duration)
```

Comment:
There's no check for overflow here, should throw an error instead of overflowing and returning an incorrect duration

Reply:
Yeah, that makes sense.